docs/standard/assembly/resolve-loads.md

```diff
@@ -1,77 +1,157 @@
+
+For information about contexts, see the <xref:System.Reflection.Assembly.LoadFrom%28System.String%29?displayProperty=nameWithType> method overload.
+
+Multiple versions of the same assembly can be loaded into the same application domain. This practice is not recommended, because it can lead to type assignment problems. See [Best practices for assembly loading](../../framework/deployment/best-practices-for-assembly-loading.md).
+
+## What the event handler should not do
+
+The primary rule for handling the <xref:System.AppDomain.AssemblyResolve> event is that you should not try to return an assembly you do not recognize. When you write the handler, you should know which assemblies might cause the event to be raised. Your handler should return null for other assemblies.
 
 > [!IMPORTANT]
-> Beginning with the .NET Framework 4, the <xref:System.AppDomain.AssemblyResolve> event is raised for satellite assemblies. This change affects an event handler that was written for an earlier version of the .NET Framework, if the handler tries to resolve all assembly load requests. Event handlers that ignore assemblies they do not recognize are not affected by this change: They return null, and normal fallback mechanisms are followed.  
+> Beginning with the .NET Framework 4, the <xref:System.AppDomain.AssemblyResolve> event is raised for satellite assemblies. This change affects an event handler that was written for an earlier version of the .NET Framework, if the handler tries to resolve all assembly load requests. Event handlers that ignore assemblies they do not recognize are not affected by this change: They return `null`, and normal fallback mechanisms are followed.
+
+When loading an assembly, the event handler must not use any of the <xref:System.AppDomain.Load%2A?displayProperty=nameWithType> or <xref:System.Reflection.Assembly.Load%2A?displayProperty=nameWithType> method overloads that can cause the <xref:System.AppDomain.AssemblyResolve> event to be raised recursively, because this can lead to a stack overflow. (See the list provided earlier in this topic.) This happens even if you provide exception handling for the load request, because no exception is thrown until all event handlers have returned. Thus, the following code results in a stack overflow if `MyAssembly` is not found:
+
+```csharp
+using System;
+using System.Reflection;
+
+class BadExample
+{
+    static void Main()
+    {
+        AppDomain ad = AppDomain.CreateDomain("Test");
+        ad.AssemblyResolve += MyHandler;
+
+        try
+        {
+            object obj = ad.CreateInstanceAndUnwrap(
+                "MyAssembly, version=1.2.3.4, culture=neutral, publicKeyToken=null",
+                "MyType");
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+    }
+
+    static Assembly MyHandler(object source, ResolveEventArgs e)
+    {
+        Console.WriteLine("Resolving {0}", e.Name);
+        return Assembly.Load(e.Name);
```

Comment:
As an extra safety item, I'd add a comment here that calling `Assembly.Load` is wrong in the handler and can cause an infinite loop.  That way, if someone copies this code without carefully reading, it gets caught.